From 9c7cef94a551cfdbbe2cce262c29221e22643d84 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ale=C5=A1=20Mat=C4=9Bj?= <amatej@redhat.com>
Date: Mon, 19 Aug 2024 06:35:03 +0200
Subject: [PATCH] fix: add missing zero initialization to region

Fixes below error:

/build/tmp-glibc/work/core2-64-wrs-linux/libtoml11/1_4.2.0-r0/git/include/toml11/fwd/../fwd/../fwd/region_fwd.hpp:49:5: error: '<unnamed>.toml::detail::region::first_' is used uninitialized [-Werror=uninitialized]
   49 |     region(region&&)      = default;
      |     ^~~~~~

Upstream-Status: Backport [https://github.com/ToruNiina/toml11/commit/9c7cef94a551cfdbbe2cce262c29221e22643d84]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 include/toml11/fwd/region_fwd.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/toml11/fwd/region_fwd.hpp b/include/toml11/fwd/region_fwd.hpp
index b40317d..26f1a9b 100644
--- a/include/toml11/fwd/region_fwd.hpp
+++ b/include/toml11/fwd/region_fwd.hpp
@@ -34,7 +34,8 @@ class region
     // a value that is constructed manually does not have input stream info
     region()
         : source_(nullptr), source_name_(""), length_(0),
-          first_line_(0), first_column_(0), last_line_(0), last_column_(0)
+          first_(0), first_line_(0), first_column_(0), last_(0), last_line_(0),
+          last_column_(0)
     {}
 
     // a value defined in [first, last).
-- 
2.34.1

